src/RhythmRunner.cpp

```diff
@@ -25,13 +25,15 @@ int main(int argc, char** argv) {
   std::shared_ptr<LevelGenerator> levelGenerator =
       std::make_shared<LevelGenerator>(music);
 
-  std::shared_ptr<Level> level = levelGenerator->generateLevel();
+  std::shared_ptr<GameRenderer> renderer = std::make_shared<GameRenderer>();
+  renderer->Init(ASSET_DIR);
 
+  std::shared_ptr<Level> level = levelGenerator->generateLevel();
   std::shared_ptr<GameState> game_state = std::make_shared<GameState>(
       level, std::make_shared<GameCamera>(), std::make_shared<Player>());
-
-  std::shared_ptr<GameRenderer> renderer = std::make_shared<GameRenderer>();
-  renderer->Init(ASSET_DIR, game_state);
+  // had to separate this since gl initialization needs to be done before
+  // generating the level
+  renderer->InitState(game_state);
```

Comment:
It looks like InitState() just sets up the keybindings. Why does this need to happen after generating the level? And why does generating the level require gl initialization?